hugr-py/src/hugr/serialization/ops.py

```diff
@@ -501,24 +497,29 @@ class OpType(RootModel):
         | AliasDefn
     ) = Field(discriminator="op")
 
+    class Config:
+        json_schema_extra = {"required": ["parent", "op"]}
+
 
 # --------------------------------------
 # --------------- OpDef ----------------
 # --------------------------------------
 
 
-class OpDef(BaseOp, populate_by_name=True):
+class FixedHugr(ConfiguredBaseModel):
+    extensions: ExtensionSet
+    hugr: Any
+
+
+class OpDef(ConfiguredBaseModel, populate_by_name=True):
     """Serializable definition for dynamically loaded operations."""
 
+    extension: ExtensionId
     name: str  # Unique identifier of the operation.
     description: str  # Human readable description of the operation.
-    inputs: list[tuple[str | None, Type]]
-    outputs: list[tuple[str | None, Type]]
-    misc: dict[str, Any]  # Miscellaneous data associated with the operation.
-    def_: str | None = Field(
-        ..., alias="def"
-    )  # (YAML?)-encoded definition of the operation.
-    extension_reqs: ExtensionSet  # Resources required to execute this operation.
+    misc: Optional[dict[str, Any]] = None
+    signature: Optional[PolyFuncType] = None
```

Comment:
```suggestion
    misc: dict[str, Any] | None = None
    signature: PolyFuncType | None = None
```

Reply:
The remaining code usues `|` so it's more consistent. Also https://docs.astral.sh/ruff/rules/non-pep604-annotation/ so I'm confused that Ruff didn't reformat it itself